### PR TITLE
Added PettingZoo support

### PR DESCRIPTION
**Pull request by Rohan138**
Changes:
New pettingzoo_env and test_pettingzoo files adding PettingZoo wrapper
Updated setup.py and requirements.txt
Fixed update_custom_actions so custom actions e.g. fire are resolved randomly over agents to eliminate bias towards agents with lower ids, similar to how update_moves resolves conflicts randomly
Ran black, isort and flake8

### CODE REVIEW

File: social_dilemmas/envs/pettingzoo_env.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        if self.num_cycles >= MAX_CYCLES:
+            self.dones = {agent: True for agent in self.agents}
+        self.agents = [agent for agent in self.agents if not self.dones[agent]]
+        return obss, rews, self.dones, infos
+
+
+class _parallel_env(ssd_parallel_env, EzPickle):
+    metadata = {"render.modes": ["human", "rgb_array"]}
+
+    def __init__(self, max_cycles, **ssd_args):
```
Comment (benblack769) [inline social_dilemmas/envs/pettingzoo_env.py:66]: This is very confusing. Some documentation somewhere (Readme, docstring or somewhere else? Not sure) to explain how to pass in arguments to SSD' pettingzoo environment would be helpful

File: social_dilemmas/envs/pettingzoo_env.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+from gym.utils import EzPickle
+from pettingzoo.utils import wrappers
+from pettingzoo.utils.conversions import from_parallel_wrapper
+from pettingzoo.utils.env import ParallelEnv
+
+from social_dilemmas.envs.env_creator import get_env_creator
+
+MAX_CYCLES = 1000
```
Comment (benblack769) [inline social_dilemmas/envs/pettingzoo_env.py:8]: Is this number standard in SSD? Where did it come from?

File: social_dilemmas/envs/pettingzoo_env.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        return self.ssd_env.render(mode=mode)
+
+    def close(self):
+        self.ssd_env.close()
+
+    def step(self, actions):
+        obss, rews, self.dones, infos = self.ssd_env.step(actions)
+        del self.dones["__all__"]
+        self.num_cycles += 1
+        if self.num_cycles >= MAX_CYCLES:
```
Comment (benblack769) [inline social_dilemmas/envs/pettingzoo_env.py:57]: If you want to check if this logic results in the same number of cycles as other pettingzoo environments (that there are no one-off counting differences), you can run the max cycles test: https://www.pettingzoo.ml/environment_creation#max-cycles-test

File: social_dilemmas/envs/pettingzoo_env.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    aec_env = wrappers.AssertOutOfBoundsWrapper(aec_env)
+    aec_env = wrappers.OrderEnforcingWrapper(aec_env)
+    return aec_env
+
+
+class ssd_parallel_env(ParallelEnv):
+    def __init__(self, env, max_cycles):
+        self.ssd_env = env
+        self.max_cycles = max_cycles
+        self.possible_agents = list(self.ssd_env.agents.keys())
```
Comment (benblack769) [inline social_dilemmas/envs/pettingzoo_env.py:31]: Might be  good to sort this, or check that the order here is reasonably deterministic.

File: social_dilemmas/envs/pettingzoo_env.py (hunk 2 above)
Comment (Rohan138) [reply social_dilemmas/envs/pettingzoo_env.py:8]: Yes, it's the 'horizon' from the RLlib implementation in run_scripts/train.py-RLlib resets environments at the end of each rollout of length 1000, so I've put in the same for SSD.

**Comment by benblack769** [comment]
Looks good to me.

**Review by eugenevinitsky — APPROVED** [review]
Thanks for the PR! Looks good to me.

**Pull request merged**